### ISSUE TITLE
Show mask only after input field is hovered or active

### ISSUE THREAD

**Issue by FiddlePlus** [uNmAnNeR/imaskjs#152]
Hey!

I'm fairly new to js and wanted to know if there is a way to integrate this plugin with default html5-placeholder (or bootstrap required placeholder)? For example, until user clicked or hovered over input field mask stays hidden and only placeholder is shown.

Would appreciate any help, thanks!

**Comment by FiddlePlus** [comment]
Here's a code i'm using. I don't think it works as i want it to work, lol. Wonder where is the mistake here..
```
<script>
window.addEventListener('DOMContentLoaded', function() {
  var showhide = document.getElementById('cheaderinput')
  showhide.addEventListener('focus', function() {
			var patternMask = new IMask(document.getElementById('cheaderinput'), {
			mask: '+{7}(000)000-00-00',
			lazy: false,  // make placeholder always visible
			placeholderChar: '_'     // defaults to '_'
			});
  }, true);
  showhide.addEventListener('blur', function() {
			var patternMask = new IMask(document.getElementById('cheaderinput'), {
			mask: '+{7}(000)000-00-00',
			lazy: true,  // make placeholder always visible
			placeholderChar: '_'     // defaults to '_'
			});
  }, true);
}, false);

</script>
```

**Comment by uNmAnNeR** [comment]
Hello! My last free minute of the day is for you :)

Try
```js
var showhide = document.getElementById('cheaderinput');
var patternMask = new IMask(showhide, {
  mask: '+{7}(000)000-00-00',
  lazy: true,  // make placeholder always visible
  placeholderChar: '_'     // defaults to '_'
});
showhide.addEventListener('focus', function() {
  patternMask.updateOptions({ lazy: false });
}, true);
showhide.addEventListener('blur', function() {
  patternMask.updateOptions({ lazy: true });
  // NEXT IS OPTIONAL
  if (!patternMask.masked.rawInputValue) {
  	patternMask.value = '';
  }
}, true);
```

It's almost always best way is to create mask once and then update options.

**Comment by FiddlePlus** [comment]
Spasibo! ;)

**Comment by FiddlePlus** [comment]
Oops, not quite there yet :(

I'm using jqBootstrapValidation.js and it tells me this - Element value was changed outside of mask. Syncronize mask using `mask.updateValue()` to work properly.

**Comment by FiddlePlus** [comment]
> Element value was changed outside of mask. Syncronize mask using `mask.updateValue()` to work properly

This issue seem to go away if i change input type from "number"  "text". But this causes validator to work incorrectly (especially on mobile). 

Is there a way to preserve input type (meaning leaving it type="number") and make it work with mask.js? Even the craziest hack will suffice ;)

**Comment by uNmAnNeR** [comment]
Not possible as i know, only text type is working correctly. What i do for number is determine platform and enable either mask or type. Seems strange, but that's web hell, where every browser on every platform defines his own specific behaviour. We are still here.
Pojaluista :)

**Comment by FiddlePlus** [comment]
Yeah, it seems not a single one of the input masking libraries is working with type-"number".

**Comment by eximius313** [comment]
@uNmAnNeR wouldn't it be better to introduce a single property (e.g. `showOnFocus`) instead of this listener thing? It's really useful feature

**Comment by uNmAnNeR** [comment]
@eximius313 Yes, it's quite often needed, even in my several project i did this trick. But still i want to keep library clean from all external dependencies as much as possible.
Ok, if i do it, then i will have to add third arg (view options) to the IMask constructor. Also someone asked about triggering lazy on hover, not focus, then i need to support also several events. Someone wants to use library with Shadow DOM, or some other elements (React Native), don't forget to SSR, and all this things tend that view should be abstracted somehow. Etc. etc...
It seems easy to implement, but only for first sight. So probably i will not, not now.

**Comment by KCH0000** [comment]
@uNmAnNeR Выполнил такой трюк, но возникла проблема, когда увожу мышь, то получаю в поле +7(7,  а если маску заполнить не полностью, например +7(926), а затем увести фокус, то получается  patternMask.value += patternMask.value, т.е. +7(926)792-6

**Comment by corsik** [comment]
> @uNmAnNeR Выполнил такой трюк, но возникла проблема, когда увожу мышь, то получаю в поле +7(7, а если маску заполнить не полностью, например +7(926), а затем увести фокус, то получается patternMask.value += patternMask.value, т.е. +7(926)792-6
Такая же проблема.

**Comment by uNmAnNeR** [comment]
@KCH0000 @corsik ребят, запилите фидл плз или что-то запускаемое, чтобы был предмет разговора. Тогда постараюсь помочь.

**Comment by eikthyrni** [comment]
@uNmAnNeR https://codepen.io/eikthyrni/pen/RwWZgOG

**Comment by chippolino** [comment]
> @uNmAnNeR Выполнил такой трюк, но возникла проблема, когда увожу мышь, то получаю в поле +7(7, а если маску заполнить не полностью, например +7(926), а затем увести фокус, то получается patternMask.value += patternMask.value, т.е. +7(926)792-6

change if
if (
                    mask.unmaskedValue.length != 11 &&
                    mask.unmaskedValue.length != 1
                ) {
                    mask.value = '';
                }

**Comment by leon9208** [comment]
Я делала так и у меня работало без проблем.

```
var inpTel = document.querySelectorAll('input[type=tel]');
var mask;
for(var i = 0; i < inpTel.length; i++) {
  inpTel[i].addEventListener('focus', function(){
    mask = IMask(this, {
        mask: '+{7} (000) 000 00 00',
        overwrite: true,
        lazy: false,
        autofix: true
    });
  })
  inpTel[i].addEventListener('blur', function(){
    if(this.value.match('_')){
      mask.masked.reset()
    }
  })
};
```
